### DISCUSSION TITLE
turn on/off battery charging based on battery %

### DISCUSSION THREAD

**Discussion by rethamazur** [humbertogontijo/homeassistant-roborock#331] [Ideas]
Hello everybody, I guess it would be nice to turn on\off the battery charging when the robot is docked.

For example, an automation could be:

- clean the entire house (vacuum only)
- when the robot is docked after the vacuuming (status attribute: charging), IF the battery is below 40%, THEN turn on charging and wait until battery reaches 55%
- segment_clean (all rooms, fan: off, mop moderate, deep+, repeats: 2)

This would prevent the robot from interrupting the second stage of the cleaning automation (mopping) due to low battery

Another nice aspect would be that we could be able to stop the charging when battery reaches 80%, that seems to be the ideal for maximizing battery life.

**Comment by Lash-L** [comment]
Unfortunately, I don't think this is something we can do as we can't control the docks charging. (If you can't do it in the Roborock app, we can't do it outside the Roborock app).

If you want to achieve this, I would look into a smart plug, but that method has its flaws

**Reply by humbertogontijo** [reply]
Even with a smart plug it would be necessary to wrap all commands in a script to check if the plug is on. Because the device will turn off after a while being unable to charge

**Reply by rethamazur** [reply]
> Unfortunately, I don't think this is something we can do as we can't control the docks charging. (If you can't do it in the Roborock app, we can't do it outside the Roborock app).
> 
> If you want to achieve this, I would look into a smart plug, but that method has its flaws

But we can control it in the Roborock app.
In the robot settings, just below the DND option, there's the "off peak charging" (don't know the exact English translation). It's used to tell the dock when to turn on the robot charging (eg: from 22 to 05)

**Reply by Lash-L** [reply]
That is not a functionality I have. What robot do you have?

**Reply by humbertogontijo** [reply]
@rethamazur Can you enable debug logs for the integration, execute the service send command with get_valley_electricity_timer and get the response in the logs? See if it matches what you have configured in the app

**Reply by rethamazur** [reply]
@Lash-L i have a S7 maxv with ultra dock.
@humbertogontijo where do I find log files? I've enabled debug logs for the integration

**Reply by humbertogontijo** [reply]
You can check it under HA web interface
Configurations -> System -> Logs

**Comment by rethamazur** [comment]
Sorry but I cannot find any Roborock related log.
I get an error: 

Logger: frontend.js.latest.202305033
Source: components/system_log/__init__.py:257
First occurred: 22:50:15 (6 occurrences)
Last logged: 23:25:51

https://...my address....ui.nabu.casa/lovelace/0:0:0 Uncaught
https://...my address...ui.nabu.casa/lovelace/0?external_auth=1:0:0 Uncaught

**Reply by Lash-L** [reply]
You have to hit view full logs.

If you don't have any logs for Roborock then you did not enable the debugging for roborock

**Reply by rethamazur** [reply]
@humbertogontijo I finally managed to get it.

2023-06-03 01:21:52.505 DEBUG (MainThread) [roborock.local_api] id=20086 Requesting method get_valley_electricity_timer with None

2023-06-03 01:21:52.519 DEBUG (MainThread) [roborock.local_api] id=20086 Response from method get_valley_electricity_timer: {'start_hour': 0, 'start_minute': 0, 'end_hour': 8, 'end_minute': 0, 'enabled': 1}

My off-peak charging timeframe is from 0.00 to 8.00, so the values are right

**Reply by Lash-L** [reply]
I can put this on my to-do list

**Reply by rethamazur** [reply]
Thanks!
